stdlib/2/formatter.pyi

```diff
@@ -0,0 +1,104 @@
+    def end_paragraph(self, blankline: int) -> None: ...
+    def add_line_break(self) -> None: ...
+    def add_hor_rule(self, *args, **kw) -> None: ...
+    def add_label_data(self, format, counter: int, blankline: int = None) -> None: ...
+    def format_counter(self, format, counter: int) -> str: ...
+    def format_letter(self, case: str, counter: int) -> str: ...
+    def format_roman(self, case: str, counter: int) -> str: ...
+    def add_flowing_data(self, data: str) -> None: ...
+    def add_literal_data(self, data: str) -> None: ...
+    def flush_softspace(self) -> None: ...
+    def push_alignment(self, align: Optional[str]) -> None: ...
+    def pop_alignment(self) -> None: ...
+    def push_font(self, font: FONT_TYPE) -> None: ...
+    def pop_font(self) -> None: ...
+    def push_margin(self, margin: int) -> None: ...
+    def pop_margin(self) -> None: ...
+    def set_spacing(self, spacing: Optional[str]) -> None: ...
+    def push_style(self, *styles: STYLES_TYPE) -> None: ...
+    def pop_style(self, n: int = 1) -> None: ...
+    def assert_line_data(self, flag: int = 1) -> None: ...
+
+class NullWriter:
+    def __init__(self) -> None: ...
+    def flush(self) -> None: ...
+    def new_alignment(self, align: Optional[str]) -> None: ...
+    def new_font(self, font: FONT_TYPE) -> None: ...
+    def new_margin(self, margin: int, level: int) -> None: ...
+    def new_spacing(self, spacing: Optional[str]) -> None: ...
+    def new_styles(self, styles) -> None: ...
+    def send_paragraph(self, blankline: int) -> None: ...
+    def send_line_break(self) -> None: ...
+    def send_hor_rule(self, *args, **kw) -> None: ...
+    def send_label_data(self, data: str) -> None: ...
+    def send_flowing_data(self, data: str) -> None: ...
+    def send_literal_data(self, data: str) -> None: ...
+
+class AbstractWriter(NullWriter):
+    def new_alignment(self, align: Optional[str]) -> None: ...
+    def new_font(self, font: FONT_TYPE) -> None: ...
+    def new_margin(self, margin: int, level: int) -> None: ...
+    def new_spacing(self, spacing: Optional[str]) -> None: ...
+    def new_styles(self, styles) -> None: ...
+    def send_paragraph(self, blankline: int) -> None: ...
+    def send_line_break(self) -> None: ...
+    def send_hor_rule(self, *args, **kw) -> None: ...
+    def send_label_data(self, data: str) -> None: ...
+    def send_flowing_data(self, data: str) -> None: ...
+    def send_literal_data(self, data: str) -> None: ...
+
+class DumbWriter(NullWriter):
+    file = ...  # type: file
+    maxcol = ...  # type: int
+    def __init__(self, file: file = None maxcol: int = 72) -> None: ...
+    def reset(self) -> None: ...
+    def send_paragraph(self, blankline: int) -> None: ...
+    def send_line_break(self) -> None: ...
+    def send_hor_rule(self, *args, **kw) -> None: ...
+    def send_literal_data(self, data) -> None: ...
+    def send_flowing_data(self, data) -> None: ...
+
+def test(file: str = None) -> None: ...
```

Comment:
But [PEP 484](https://www.python.org/dev/peps/pep-0484/#union-types) says:

An optional type is also automatically assumed when the default value is `None` , for example:

    def handle_employee(e: Employee = None): ...

This is equivalent to:

    def handle_employee(e: Optional[Employee] = None) -> None: ...